edx_notifications/lib/consumer.py

```diff
@@ -0,0 +1,121 @@
+"""
+All in-proc API endpoints for acting as a Notifications Consumer
+NOTE that we can only query for notifications that are "durable", i.e.
+persisted in our database
+"""
+
+import pytz
+from datetime import datetime
+
+from edx_notifications.stores.store import notification_store
+
+from edx_notifications.exceptions import (
+    ItemIntegrityError
+)
+
+
+def get_notifications_count_for_user(user_id, filters=None):
+    """
+    Returns the number of notifications this user has
+
+    ARGS:
+        user_id: The id of the user to query
+        filters: A dict containing the following optional keys
+            - 'namespace': what namespace to search (defuault None)
+            - 'read': Whether to return read notifications (default True)
+            - 'unread': Whether to return unread notifications (default True)
+            - 'type_name': what msg_types to count
+
+    NOTE: If the 'msg_id' filter is passed in, a ValueError will be raised as
+    this is not a valid combination
+    """
```

Comment:
I can see your point. I started down this path because of pylint starting to complain about too many optional arguments :-)

That said, I think having a schema'd Filter class make sense.